tools/gyp/gyp/MSVS/MSVSVersion.py

```diff
@@ -0,0 +1,366 @@
+# Copyright (c) 2013 Google Inc. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+"""Handle version information related to Visual Stdio."""
+
+import os
+import glob
+from typing import Dict
+from gyp.MSVS.MSVSUtil import TryQueryRegistryValue
+
+msvs_version_map = {
+  'auto': ('16.0', '15.0', '14.0', '12.0', '10.0', '9.0', '8.0', '11.0'),
```

Comment:
Maybe lets keep 2017 as the default for now, until we get it tested in CI?

Reply:
That's the GYP way...
For Node.js we need to make sure GYP3 isn't called with `auto`:
https://github.com/nodejs/node/blob/1f1914c9d454875670a48ac943e98d78671b20c2/vcbuild.bat#L175
https://github.com/nodejs/node/blob/1f1914c9d454875670a48ac943e98d78671b20c2/vcbuild.bat#L259